### PR TITLE
EmailValidator with IDN fails on hyphen

### PR DESCRIPTION
**Pull request by mapeveri**
| Q             | A
| ------------- | ---
| Is bugfix?    | ✔️
| New feature?  | ❌
| Breaks BC?    | ❌
| Fixed issues  | #2

### CODE REVIEW

**Comment by samdark** [comment]
It seems to break tests: https://github.com/yiisoft/validator/runs/1438130363

File: src/Rule/Email.php [hunk 1]
```diff
@@ -93,6 +94,10 @@ protected function validateValue($value, DataSetInterface $dataSet = null): Resu
             }
         }
 
+        if ($this->enableIDN && $valid === false) {
+            $valid = filter_var($originalValue, FILTER_VALIDATE_EMAIL) ? true : false;
```
Comment (samdark) [inline src/Rule/Email.php:98]:
```suggestion
            $valid = (bool) filter_var($originalValue, FILTER_VALIDATE_EMAIL);
```

**Comment by mapeveri** [comment]
@samdark Yep, the problem is with this emails:

image

I commeted the email with problems in the test. Any idea why this email passed the filter_var?

Thanks

File: src/Rule/Email.php (hunk 1 above)
Comment (mapeveri) [reply src/Rule/Email.php]: I updated this fix

**Comment by samdark** [comment]
@mapeveri seems that filter_var isn't suitable for the purpose: https://3v4l.org/4oslP

**Comment by samdark** [comment]
filter_var is RFC 5321 and we need way stricter validation.

**Comment by mapeveri** [comment]
@samdark What validation do you recommend?

**Comment by samdark** [comment]
Regular expressions.

**Comment by mapeveri** [comment]
@samdark thanks. I updated this PR :)

File: src/Rule/Email.php [hunk 2]
```diff
@@ -93,6 +94,11 @@ protected function validateValue($value, DataSetInterface $dataSet = null): Resu
             }
         }
 
+        if ($this->enableIDN && $valid === false) {
+            $pattern = '/^([a-zA-Z0-9._%+-]+)@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.)|(([\w-]+\.)+))([a-zA-Z]{2,4}|[0-9]{1,3})(\]?)$/';
```
Comment (samdark) [inline src/Rule/Email.php:98]: That could be moved to a customizable property same as other regex patterns.

Comment (mapeveri) [reply src/Rule/Email.php:98]: @samdark there is a private pattern regex, maybe a new property $patternIdnEmail as a public property, is ok?

Comment (mapeveri) [reply src/Rule/Email.php]: @samdark i upated the PR ;)

Comment (mapeveri) [reply src/Rule/Email.php]: @samdark is it ok?

**Review by samdark — APPROVED** [review]

**Comment by samdark** [comment]
Thank you!

File: src/Rule/Email.php (hunk 2 above)
Comment (samdark) [reply src/Rule/Email.php]: Yes.

**Pull request merged**